packages/flutter/lib/src/material/scrollbar.dart

```diff
@@ -2,236 +2,205 @@
+    required Widget child,
+    ScrollController? controller,
+    bool isAlwaysShown = false,
+    this.showTrackOnHover = false,
+    this.thicknessWithTrack,
+    double? thickness,
+    Radius? radius,
+  }) : super(
+    key: key,
+    child: child,
+    controller: controller,
+    isAlwaysShown: isAlwaysShown,
+    thickness: thickness ?? _kScrollbarThickness,
+    radius: radius,
+    fadeDuration: _kScrollbarFadeDuration,
+    timeToFade: _kScrollbarTimeToFade,
+    pressDuration: Duration.zero,
+  );
+
+  /// Controls if the track will show on hover and remain, including during drag.
   ///
-  /// Typically a [ListView] or [CustomScrollView].
-  final Widget child;
+  /// Defaults to false, cannot be null.
+  final bool showTrackOnHover;
 
-  /// {@macro flutter.cupertino.cupertinoScrollbar.controller}
-  final ScrollController? controller;
-
-  /// {@macro flutter.cupertino.cupertinoScrollbar.isAlwaysShown}
-  final bool isAlwaysShown;
-
-  /// The thickness of the scrollbar.
-  ///
-  /// If this is non-null, it will be used as the thickness of the scrollbar on
-  /// all platforms, whether the scrollbar is being dragged by the user or not.
-  /// By default (if this is left null), each platform will get a thickness
-  /// that matches the look and feel of the platform, and the thickness may
-  /// grow while the scrollbar is being dragged if the platform look and feel
-  /// calls for such behavior.
-  final double? thickness;
-
-  /// The radius of the corners of the scrollbar.
+  /// The thickness of the scrollbar when a hover state is active and
+  /// [showTrackOnHover] is true.
   ///
-  /// If this is non-null, it will be used as the fixed radius of the scrollbar
-  /// on all platforms, whether the scrollbar is being dragged by the user or
-  /// not. By default (if this is left null), each platform will get a radius
-  /// that matches the look and feel of the platform, and the radius may
-  /// change while the scrollbar is being dragged if the platform look and feel
-  /// calls for such behavior.
-  final Radius? radius;
+  /// Defaults to 12.0 dp when null.
+  final double? thicknessWithTrack;
 
   @override
   _ScrollbarState createState() => _ScrollbarState();
 }
 
-class _ScrollbarState extends State<Scrollbar> with SingleTickerProviderStateMixin {
-  ScrollbarPainter? _materialPainter;
-  late TextDirection _textDirection;
-  late Color _themeColor;
-  late bool _useCupertinoScrollbar;
-  late AnimationController _fadeoutAnimationController;
-  late Animation<double> _fadeoutOpacityAnimation;
-  Timer? _fadeoutTimer;
+class _ScrollbarState extends RawScrollbarState<Scrollbar> {
+  late AnimationController _hoverAnimationController;
+  bool _dragIsActive = false;
+  bool _hoverIsActive = false;
+
+  Set<MaterialState> get _states => <MaterialState>{
+    if (_dragIsActive) MaterialState.dragged,
+    if (_hoverIsActive) MaterialState.hovered,
+  };
+
+  MaterialStateProperty<Color> get _thumbColor {
```

Comment:
The default values of the _thumbColor, _trackColor, etc properties should be based on the theme's color scheme because dark mode and consistency with other Material components.

Reply:
Updated! :)